services/motion/builtin/move_request.go

```diff
@@ -305,6 +301,47 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 	return state.ExecuteResponse{}, nil
 }
 
+func (mr *moveRequest) augmentBaseExecutionState(
+	baseExecutionState motionplan.ExecutionState,
+) (motionplan.ExecutionState, error) {
+	// update plan
+	existingPlan := baseExecutionState.Plan()
+	newTrajectory := make(motionplan.Trajectory, 0, len(existingPlan.Trajectory()))
+	for idx, currTraj := range existingPlan.Trajectory() {
+		if idx != 0 {
+			idx--
+		}
+		currPathStep := existingPlan.Path()[idx]
+		kbPose := currPathStep[mr.kinematicBase.Kinematics().Name()]
+		kbTraj := currTraj[mr.kinematicBase.Name().Name]
+		updatedTraj := kbTraj
+		updatedTraj = append(updatedTraj, referenceframe.PoseToInputs(kbPose.Pose())...)
+		newTrajectory = append(
+			newTrajectory, map[string][]referenceframe.Input{mr.kinematicBase.Kinematics().Name(): updatedTraj},
+		)
+	}
+	augmentedPlan := motionplan.NewSimplePlan(existingPlan.Path(), newTrajectory)
+
+	// update currentInputs
+	allCurrentInputsFromBaseExecutionState := baseExecutionState.CurrentInputs()
+	kinematicBaseCurrentInputs := allCurrentInputsFromBaseExecutionState[mr.kinematicBase.Kinematics().Name()]
+	kinematicBaseCurrentInputs = append(
+		kinematicBaseCurrentInputs,
+		referenceframe.PoseToInputs(baseExecutionState.CurrentPoses()[mr.kinematicBase.LocalizationFrame().Name()].Pose())...,
+	)
+	allCurrentInputsFromBaseExecutionState[mr.kinematicBase.Kinematics().Name()] = kinematicBaseCurrentInputs
+
+	// update currentPoses
+	existingCurrentPoses := baseExecutionState.CurrentPoses()
+	localizationFramePose := existingCurrentPoses[mr.kinematicBase.LocalizationFrame().Name()]
+	existingCurrentPoses[mr.kinematicBase.Name().Name] = localizationFramePose
+	delete(existingCurrentPoses, mr.kinematicBase.LocalizationFrame().Name())
```

Comment:
We
```
delete(existingCurrentPoses, mr.kinematicBase.LocalizationFrame().Name())
```
here since the name of the wrapperFrame is the same as the name of the kinematicBase.Kinematics.
There is no need for us to have information pertaining to the localizationFrame.

Reply:
Sure; I meant, perhaps the localization frame need not have a different name.

Reply:
I would prefer to have the localizationFrame use a separate name since it functions slightly differently than the execution (aka the PTG) frame.
I personally think it would be a little more confusing to reason about the localizationFrame and how it should be used and when should it have to same name as the executionFrame.